provision/acc_provision/acc_provision.py

```diff
@@ -2019,58 +2019,104 @@ def has_multi_subnet(config):
     return True
 
 
-def is_dualstack_config(config):
-    if "net_config" not in config:
-        return False
-    subnet_info = []
-    if not isinstance(config["net_config"]["pod_subnet"], list):
-        subnet_info[-1] = config["net_config"]["pod_subnet"]
-    else:
-        pod_subnets = []
-        for pod_subnet in config["net_config"]["pod_subnet"]:
-            pod_subnets.append(pod_subnet)
-        for pod_subnet in pod_subnets:
-            subnet_info.append(pod_subnet)
+def is_support_dualstack(flavor):
+    version = flavor.split("-")[1]
+    support_k8s_version = "1.21"
+    support_openshift_version = "4.8"
+    if version >= support_k8s_version:
+        return True
 
-    if not isinstance(config["net_config"]["node_subnet"], list):
-        subnet_info[-1] = config["net_config"]["node_subnet"]
-    else:
-        node_subnets = []
-        for node_subnet in config["net_config"]["node_subnet"]:
-            node_subnets.append(node_subnet)
-        for node_subnet in node_subnets:
-            subnet_info.append(node_subnet)
+    if version >= support_openshift_version:
+        return True
 
-    if not isinstance(config["net_config"]["extern_dynamic"], list):
-        subnet_info[-1] = config["net_config"]["extern_dynamic"]
+    return False
+
+
+def process_subnet_value(subnet_value, subnet_info):
+    if not isinstance(subnet_value, list):
+        subnet_info.append(subnet_value)
     else:
-        extern_dynamics = []
-        for extern_dynamic in config["net_config"]["extern_dynamic"]:
-            extern_dynamics.append(extern_dynamic)
+        for subnet in subnet_value:
+            subnet_info.append(subnet)
 
-        for extern_dynamic in extern_dynamics:
-            subnet_info.append(extern_dynamic)
+
+def get_subnet_list(config):
+    subnet_info = []
+    if "net_config" not in config:
+        return subnet_info
+
+    net_config = config["net_config"]
+    process_subnet_value(net_config.get("pod_subnet", []), subnet_info)
+    process_subnet_value(net_config.get("node_subnet", []), subnet_info)
+    process_subnet_value(net_config.get("extern_dynamic", []), subnet_info)
+    process_subnet_value(net_config.get("extern_static", []), subnet_info)
+
+    return subnet_info
+
+
+def is_dualstack_config(config):
+    subnet_info = get_subnet_list(config)
+    subnet_type = determine_subnet_type(subnet_info)
+    if subnet_type == "DualStack":
+        return True
+    return False
+
+
+def determine_subnet_type(subnet_info):
+    if len(subnet_info) == 0:
+        return ""
+
+    has_ipv4 = False
+    has_ipv6 = False
 
     for subnet in subnet_info:
         rtr, _ = subnet.split("/")
         ip = ipaddress.ip_address(rtr)
```

Comment:
what if the address malformed, this conversion will throw an exception and acc-provision will exit with a stack trace?

Reply:
addressed

Reply:
how? I still don't see a try/catch block...I see a check for empty string, but the ip address could be malformed for other, reasons, no?